compiler/src/iree/compiler/Utils/ElementPackingUtils.h

```diff
@@ -27,6 +33,11 @@ bool needToPackSubByteElements(RankedTensorType shapedType);
 /// runtime and kernel. For such cases, we perform tight packing for supported
 /// sub-byte elements, and expand to the next power-of-two bit width for other
 /// cases.
+Type legalizeStorageElementType(Type elementType, bool isPackedStorage);
```

Comment:
bool args are an antipattern we try to avoid - see e.g. https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/#4-otherwise-split-the-function

plumbing the bool down through the encoding utils also makes this much more difficult to follow (in what is already difficult code). the decision should be made at the top level or all the way at the bottom by checking the encoding. AFAICT the reason for the spaghetti here is the flag: remove the flag or make it harder to use, as the flag is temporary but the code is going to live much longer and needs to be cleaner.